tools/points_association_nn/points_association_nn.xml

```diff
@@ -0,0 +1,41 @@
+<tool id="ip_points_association_nn" name="Association of points" version="0.0.1" profile="20.05"> 
+    <description>in consecutive frames (slices) using the nearest neighbor algorithm</description>
+    <requirements>
+        <requirement type="package" version="1.20.2">numpy</requirement>
+        <requirement type="package" version="1.2.4">pandas</requirement>
+        <requirement type="package" version="0.18.1">scikit-image</requirement>
+    </requirements>
+    <command>
+    <![CDATA[
+         python '$__tool_directory__/points_association_nn.py'
+         '$fn_in'
+         '$fn_out'
+         '$nbpx'
+         '$thres'
+         '$minlen'
+    ]]>
+    </command>
+    <inputs>
+        <param name="fn_in" type="data" format="tabular" label="Name of input file (tsv tabular)" />
+        <param name="nbpx" type="integer" value="6" label="Neighborhood size in pixel" />
+        <param name="thres" type="float" value="25" label="Percentage (%) of the global maximal intensity for thresholding some event" />
+        <param name="minlen" type="float" value="50" label="Minimum length of tracks (% of sequence length)" />
+    </inputs>
+    <outputs>
+        <data format="xlsx" name="fn_out" />
```

Comment:
Any reason why we should create xls files?

Reply:
> Any reason why we should create xls files?

I want to save data to an xlsx with multiple sheets as I have to output a 3D array. I also need to keep the data structure simple so that it is easy for other tools to load it.

Reply:
... and the data saved must be human-readable .

Reply:
Is that a manual step in the workflow? The problem is that no tools in Galaxy can deal with xls, so you can not use that tool in a workflow. Usually people use sqlite or hdf5 or netcdf for arrays.